ros_packages/motors/motors/motor_control.py

```diff
@@ -5,137 +5,149 @@
-                        self.get_logger().warn(f"Error while processing joint-trajectory-message: {str(e)}")
-        
 
+    def __init__(self):
+
+        qos_policy = rclpy.qos.QoSProfile(
+            reliability=rclpy.qos.ReliabilityPolicy.BEST_EFFORT,
+            history=rclpy.qos.HistoryPolicy.KEEP_LAST,
+            depth=1
+        )
+
+        super().__init__('motor_control')
+
+        # Toggle Devmode
+        self.declare_parameter("dev", False)
+        self.dev = self.get_parameter("dev").value
+
+        # Topic for JointTrajectory
+        self.subscription = self.create_subscription(
+            JointTrajectory,
+            'joint_trajectory',
+            self.joint_trajectory_callback,
+            qos_profile=qos_policy
+        )
+
+        # Service for MotorSettings
+        self.srv = self.create_service(
+            MotorSettingsSrv,
+            'motor_settings',
+            self.motor_settings_callback
+        )
+
+        # Publisher for MotorSettings
+        self.publisher = self.create_publisher(
+            MotorSettings, "motor_settings", 10)
+
+        # load motor-settings if not in dev mode
+        if not self.dev:
+            for motor in motors:
+                if self.check_if_motor_is_connected(motor):
+                    successful, motor_settings_dto = motor_client.get_motor_settings(
+                        motor.name)
+                    if successful:
+                        motor.apply_settings(motor_settings_dto)
+
+        # get UID from database
+        response = update_bricklet_uids.get_uids_from_db()
+        UID1 = response[0]
+        UID2 = response[1]
+        UID3 = response[2]
+
+        # Log that initialization is complete
+        self.get_logger().warn("Info: passed __init__")
+
+    # Check if motor is connected
+    # Non-connected bricklet UIDs seem to be only 2 characters long
+    # Connected ones at least 6 characters long
+    def check_if_motor_is_connected(self, motor):
+        for bricklet_pin in motor.bricklet_pins:
+            # X,Y and Z are the default uid of a Servo Bricklet 2.0 (updated ob boot by update_bricklet_uids.py)
+            if bricklet_pin.bricklet.uid_string != 'X' and bricklet_pin.bricklet.uid_string != 'Y' and bricklet_pin.bricklet.uid_string != 'Z':
+                return True
+        return False
+
+    def motor_settings_callback(self, request: MotorSettingsSrv.Request, response: MotorSettingsSrv.Response):
+
+        response.settings_applied = True
+        response.settings_persisted = True
+
+        motor_settings_ros = request.motor_settings
+        motor_settings_dto = motor_settings_ros_to_dto(motor_settings_ros)
+
+        try:
+            motors = name_to_motors[request.motor_settings.motor_name]
+            for motor in motors:
+                motor_settings_dto['name'] = motor.name
+                motor_settings_ros.motor_name = motor.name
+                applied = motor.apply_settings(motor_settings_dto)
```

Comment:
shouldn't we also check here, if we are actually connected to a bricklet?

Reply:
Is done